.buildkite/scripts/steps/build_and_test.sh

```diff
@@ -75,5 +75,10 @@ if ! [[ "$HARDWARE_ARCH" = aarch64 && -z "$CPP_CROSS_COMPILE" ]] ; then
 fi
 
 if ! [[ "$HARDWARE_ARCH" = aarch64 && -n "$CPP_CROSS_COMPILE" ]] ; then 
-  buildkite-agent artifact upload "build/distributions/*;*/**/ml_test_*.out"
+  buildkite-agent artifact upload "build/distributions/*.zip"
+fi
+
+if [[ -z "$CPP_CROSS_COMPILE" ]] ; then 
+  find . -path  "*/**/ml_test_*.out" -o -path "*/**/*.junit" | xargs tar cvzf unit_test_results.tgz
+  buildkite-agent artifact upload "unit_test_results.tgz"
```

Comment:
Will these files from different platforms overwrite each other?  If `linux-aarch64` and `linux-x86_64` both upload test results will we just be left with whichever was uploaded last? If so, please include the platform name in the file name (and do the same for Windows for consistency even though it's a zip).

Reply:
No, they won't overwrite each other as they're uploaded from different steps, and each step has it's own unique artifact location. I'll give them all unique names for ease of use when downloading etc.